scripts/xmpro-website-scripts/scrape-xmpro-website-usecases.py

```diff
@@ -0,0 +1,147 @@
+        if response.status_code == 200:
+            # Parse the HTML content
+            soup = BeautifulSoup(response.text, "html.parser")
+            
+            # Find the title of the page
+            title_elem = soup.find("title")
+            if title_elem:
+                title = title_elem.text.strip()
+
+                # Find the portfolio-inner div
+                portfolio_inner_div = soup.find("div", class_="portfolio-inner")
+
+                if portfolio_inner_div:
+                    # Find all content elements within the portfolio-inner div
+                    content_elements = portfolio_inner_div.find_all(["p", "h1", "h2", "h3", "h4", "img"])
+
+                    # Initialize markdown content
+                    markdown_content = ""
+
+                    # Append content to the markdown content preserving order
+                    for element in content_elements:
+                        if element.name == "img":
+                            prop = "src"
+                            if "data-src" in element.attrs:
+                                prop = "data-src"
+                            w = element.get("width")
+                            h = element.get("height")
+                            markdown_content += f'<img src="{element[prop]}" width="{w}" height="{h}">\n\n'
+                        else:
+                            if element.name.startswith("h"):
+                                heading_level = int(element.name[1])
+                                markdown_content += f"{'#' * heading_level} {element.get_text(strip=True)}\n\n"
+                            else:
+                                markdown_content += f"{element.get_text(strip=True)}\n\n"
+
+                    # Save content to a Markdown file and return filename with title
+                    return save_to_md(title, markdown_content, url, folder_path)
+                else:
+                    print("Portfolio inner div not found.")
+            else:
+                print("Title element not found.")
+        else:
+            print(f"Failed to retrieve page {url}. Status code: {response.status_code}")
+    except Exception as e:
+        print(f"Error occurred while scraping page {url}: {e}")
+    return None
+
+# Function to update or create README.md with hyperlinks to exported markdown files
+def update_readme(folder_path, md_files, title):
+    try:
+        readme_file = os.path.join(folder_path, "copy-me-use-cases.md")
+        with open(readme_file, 'w', encoding='utf-8') as file:
+            file.write(f"# {title}\n\n")
+            for md_file in md_files:
+                # Get the filename without the folder path
+                filename = os.path.basename(md_file['filename'])
+                # Get the folder path location without the "docs/" part and replace "\" with "/"
+                formatted_filename = md_file['filename'].replace("docs/", "").replace("\\", "/")
```

Comment:
@jacksonmcguirkxmpro 

Instead of find replace - lets turn this into a `Path` object and then get the correct path - 

We can run an exercise on this - just let me know when you are free

Reply:
resoleved all scripts with path.as_posix(_) and path.relative("docs/")